### PR TITLE
Initial updates to v2, linear actuator replaced with valve

### PR DESCRIPTION
**Pull request by Oasixer**
UPDATE: It compiles now, (at least on my machine)

---
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4)

### CODE REVIEW

**Review by jacobdeery — COMMENTED** [review]
> Reviewed 1 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4)* status: 1 of 9 files reviewed, 6 unresolved discussions (waiting on @Oasixer and @zx100x100)

___
*[main.c, line 18 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdU9BwEDadJwM0Sp1k:-MCdU9BwEDadJwM0Sp1l:b-dejtz7) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/main.c#L18)):*
> ```
> /* #include "lin_actuator.h" */
> ```
Remove
___
*[main.c, line 85 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdUBcG4tJOeASUsAG_:-MCdUBcG4tJOeASUsAGa:b-1kqy5w) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/main.c#L85)):*
<details><summary><i alt="summary">Quoted 9 lines of code…</i></summary>

> ```
>     /* while (!lin_actuator_dac_init()) { */
>         /* if (millis() - last_millis > MAX_LOOP_TIME_DIFF_ms) { */
>             /* can_msg_t error_msg; */
>             /* build_board_stat_msg(millis(), E_CANNOT_INIT_DACS, NULL, 0, &error_msg); */
>             /* txb_enqueue(&error_msg); */
>             /* last_millis = millis(); */
>         /* } */
>         /* txb_heartbeat(); // send out queued CAN messages */
>     /* } */
> ```
</details>

Remove
___
*[valve.h, line 2 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdUEn96o_4P1zTJTsB:-MCdUEn96o_4P1zTJTsC:b72nxyk) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/valve.h#L2)):*
> ```objc
> #ifndef VALVE_H
> #define	VALVE_H
> ```

Fix spacing?
___
*[valve.h, line 15 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdUIATBkaWiS2DDB3h:-MCdUIATBkaWiS2DDB3i:beaik9m) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/valve.h#L15)):*
> ```
> bool getValveOpenState();
> bool getValveClosedState();
> ```
These functions should have `void` as params
___
*[valve.c, line 24 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdULwi7i7VehG6h4bj:-MCdULwi7i7VehG6h4bk:b-vatvcg) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/valve.c#L24)):*
> ```
> ····
> ```
remove leading whitespace
___
*[valve.c, line 39 at r1](https://reviewable.io/reviews/waterloo-rocketry/cansw_vent/4#-MCdUO8yCySAzKorfE7s:-MCdUO8yCySAzKorfE7t:b-d075z9) ([raw file](https://github.com/waterloo-rocketry/cansw_vent/blob/ed00680e0f0ae5011c5bee8a04f6884dd830bb77/valve.c#L39)):*
> ```C
> }
> 
> // TODO: do I really need to use snake_case in C? can I just refactor this entire codebase into camelCase?
> ```

I don't have strong feelings one way or another, but please make these functions `snake_case` for now and then refactor in a separate PR if you plan to do so. Keeping things consistent is always best

**Comment by Oasixer** [comment]
@jacobdeery for some reason with the way that your comments are left, I can't find a way to respond to each comment.

Anyway, I fixed all your suggestions, TY for the review.

@zx100x100
This should be finalized - or as much as I can finalize it without Zach's feedback RE: whether it actually works

File: README.md [hunk 1]
```diff
@@ -9,3 +9,15 @@ Runs on the vent PCB (GYOZA). To load code onto the board:
...
+#### Notes
+Just Kaelan's (who is updating this codebase for vent board v2) notes on requirements for the updates.
+
+* We have gone from needing a linear actuator to control the valve to the valve being a solenoid
+  * Therefore we dont get any feedback from the valve
+  * And we use a pin that goes high or low to activate the solenoid instead of whatever complicated controls were needed for linac
+
+* look @ schematics, make it do the same thing as before.
+
+* there are some voltage dividers for a limit switch, those are so we can use board for moar, not needed in code.
```
Comment (zx100x100) [inline README.md]: There are pull up resistors, not voltage dividers, more is misspelled.

File: main.c [hunk 2]
```diff
@@ -15,7 +15,7 @@
```
Comment (zx100x100) [inline main.c:14]: On line 11 you have #include "mcc_generated_files/i2c1.h". nothing on the new board uses I2C any more. you should delete this line, along with the associated files, and the call to i2c1_init() on line 44

File: main.c [hunk 3]
```diff
@@ -69,19 +69,12 @@ int main(int argc, char** argv) {
...
-    while (!lin_actuator_dac_init()) {
-        if (millis() - last_millis > MAX_LOOP_TIME_DIFF_ms) {
-            can_msg_t error_msg;
-            build_board_stat_msg(millis(), E_CANNOT_INIT_DACS, NULL, 0, &error_msg);
-            txb_enqueue(&error_msg);
-            last_millis = millis();
-        }
-        txb_heartbeat(); // send out queued CAN messages
-    }
     RED_LED_OFF();
```
Comment (zx100x100) [inline main.c]: you turn the red LED on, then off immediately, this is weird and useless, I would just delete both lines for now.

File: error_checks.c [hunk 4]
```diff
@@ -101,11 +101,3 @@ bool check_valve_pin_error(enum VALVE_STATE req_state) {
     // check that digital pin reads match what they should be (a la injector)
```
Comment (zx100x100) [inline error_checks.c:101]: bool check_valve_pin_error(enum VALVE_STATE req_state) just returns true right now, at absolute minimum a TODO comment should be added, preferably just implement it.

File: valve.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "canlib/can.h"
+#include "canlib/can_common.h"
+#include "canlib/pic18f26k83/pic18f26k83_can.h"
+#include "canlib/message_types.h"
+#include "canlib/util/can_tx_buffer.h"
+
+#include "mcc_generated_files/i2c1.h"
```
Comment (zx100x100) [inline valve.c]: delete this, we don't use i2c any more.

**Review by zx100x100 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (Oasixer) [reply README.md]: Fixed

File: main.c (hunk 3 above)
Comment (Oasixer) [reply main.c]: Gotcha

File: error_checks.c (hunk 4 above)
Comment (Oasixer) [reply error_checks.c:101]: adding TODO for now

File: valve.c [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+#include "mcc_generated_files/mcc.h"
+
+#include "valve.h"
+#include "vent.h"
+#include "timer.h"
+
+void valve_init(void){
+    TRISB5 = 0; // set VALVE_CONTROL (pin 26) as output
+    LATB5 = 0; // this opens the valve, which is what we want on startup
+
```
Comment (zx100x100) [inline valve.c:20]: the LED pins are wrong for all 3 LEDs, please fix them based on the schematic

**Review by zx100x100 — APPROVED** [review]
Tested it and it works well

**Review by AidanHa — APPROVED** [review]

**Pull request merged**